### PR TITLE
update dockerfile to include engine and build from scratch

### PR DESCRIPTION
**Pull request by dlsteuer**

### CODE REVIEW

**Comment by dlsteuer** [comment]
This fixes battlesnakeio/roadmap#56 also an automated build has been setup on docker hub, so that on merges to master, docker hub will pull and build an image.

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
 COPY . .
 RUN CGO_ENABLED=0 GOOS=linux go install -installsuffix cgo ./cmd/...
+RUN CGO_ENABLED=0 GOOS=linux go install -installsuffix cgo .
 
-FROM alpine:3.7
+FROM alpine:latest as certs
 RUN apk add --no-cache ca-certificates
-COPY --from=0 /go/bin/ /bin/
+
+FROM scratch
```
Comment (ColDog) [inline Dockerfile]: I'd prefer to use `alpine`, scratch is cool but if you need to just exec in and look at the filesystem you don't even have a basic shell installed. It's not worth it for 1 - 2 mb of savings over alpine I think. @dlsteuer

Comment (dlsteuer) [reply Dockerfile]: You make good points, but it was *5 whole MB*

Comment (ColDog) [reply Dockerfile]: lol, let's try it I guess, see what happens.

**Pull request merged**